pkg/appendable/index_file_csv_test.go

```diff
@@ -122,4 +162,95 @@ func TestAppendDataRowCSV(t *testing.T) {
 
 	})
 
+	// this test case fails due to the way we measure checksums
+	// since CSV's are column-based appending new columns, cause checksums and endbyteoffsets to be mismatched
+	t.Run("new index", func(t *testing.T) {
+		i, err := NewIndexFile(CSVHandler{ReadSeeker: strings.NewReader("header1\ntest1\n")})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		buf := &bytes.Buffer{}
+
+		if err := i.Serialize(buf); err != nil {
+			t.Fatal(err)
+		}
+
+		j, err := ReadIndexFile(buf, CSVHandler{ReadSeeker: strings.NewReader("header1,header2\ntest1,test2\n")})
```

Comment:
this comment will probably get lost in the ether, so I'll make sure to document this. But for CSV we'll assume this never happens. 

That is, appending a new column.